src/openrct2/platform/Platform.Win32.cpp

```diff
@@ -347,6 +351,129 @@ namespace Platform
         return reinterpret_cast<uintptr_t>(input);
     }
 
+    void SetUpFileAssociations()
+    {
+        // Setup file extensions
+        SetUpFileAssociation(".sc4", "RCT1 Scenario (.sc4)", "Play", "\"%1\"", 0);
+        SetUpFileAssociation(".sc6", "RCT2 Scenario (.sc6)", "Play", "\"%1\"", 0);
+        SetUpFileAssociation(".sv4", "RCT1 Saved Game (.sc4)", "Play", "\"%1\"", 0);
+        SetUpFileAssociation(".sv6", "RCT2 Saved Game (.sv6)", "Play", "\"%1\"", 0);
+        SetUpFileAssociation(".sv7", "RCT Modified Saved Game (.sv7)", "Play", "\"%1\"", 0);
+        SetUpFileAssociation(".td4", "RCT1 Track Design (.td4)", "Install", "\"%1\"", 0);
+        SetUpFileAssociation(".td6", "RCT2 Track Design (.td6)", "Install", "\"%1\"", 0);
+
+        // Refresh explorer
+        SHChangeNotify(SHCNE_ASSOCCHANGED, SHCNF_IDLIST, nullptr, nullptr);
+    }
+
+    static HMODULE _dllModule = nullptr;
+    static HMODULE GetDLLModule()
+    {
+        if (_dllModule == nullptr)
+        {
+            _dllModule = GetModuleHandle(nullptr);
+        }
+        return _dllModule;
+    }
+
+    static std::wstring get_progIdName(const std::string_view& extension)
+    {
+        auto progIdName = std::string(OPENRCT2_NAME) + std::string(extension);
+        auto progIdNameW = String::ToWideChar(progIdName);
+        return progIdNameW;
+    }
+
+    bool SetUpFileAssociation(
+        const std::string extension, const std::string fileTypeText, const std::string commandText,
+        const std::string commandArgs, const uint32_t iconIndex)
+    {
+        wchar_t exePathW[MAX_PATH];
+        wchar_t dllPathW[MAX_PATH];
+
+        [[maybe_unused]] int32_t printResult;
+
+        GetModuleFileNameW(nullptr, exePathW, (DWORD)std::size(exePathW));
+        GetModuleFileNameW(GetDLLModule(), dllPathW, (DWORD)std::size(dllPathW));
+
+        auto extensionW = String::ToWideChar(extension);
+        auto fileTypeTextW = String::ToWideChar(fileTypeText);
+        auto commandTextW = String::ToWideChar(commandText);
+        auto commandArgsW = String::ToWideChar(commandArgs);
+        auto progIdNameW = get_progIdName(extension);
+
+        HKEY hKey = nullptr;
+        HKEY hRootKey = nullptr;
+
+        // [HKEY_CURRENT_USER\Software\Classes]
+        if (RegOpenKeyW(HKEY_CURRENT_USER, SOFTWARE_CLASSES, &hRootKey) != ERROR_SUCCESS)
```

Comment:
Technically these functions shouldn't be used all the reg functions should be using the exw versions. But that's for a future refactor.